llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -2313,6 +2313,26 @@ bool TargetLowering::SimplifyDemandedBits(
     Known = TLO.DAG.computeKnownBits(Op, DemandedElts, Depth);
     break;
   }
+  case ISD::SINT_TO_FP: {
+    EVT InnerVT = Op.getOperand(0).getValueType();
+    const APInt &DemandedBitsToUse =
+        TLO.DAG.getTargetLoweringInfo().isBitcastFree(VT, InnerVT)
+            ? OriginalDemandedBits
```

Comment:
You can't safely use OriginalDemandedBits as it won't account for multiple uses - you will need to move this inside SimplifyMultipleUseDemandedBits

Reply:
Fixed.